serviceerror/namespace_unavailable.go

```diff
@@ -0,0 +1,80 @@
+// The MIT License
+//
+// Copyright (c) 2024 Temporal Technologies Inc.  All rights reserved.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package serviceerror
+
+import (
+	"fmt"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"go.temporal.io/api/errordetails/v1"
+)
+
+type (
+	// NamespaceUnavailable is returned by the service when a request addresses a namespace that is unavailable. For
+	// example, when a namespace is in the process of failing over between clusters. This is a transient error that
+	// should be automatically retried by clients.
+	NamespaceUnavailable struct {
+		Namespace string
+		// Note that other service errors have a private status.Status field, there's no compelling reason to do
+		// copy that pattern here. The status can and should be computed on the fly in case the
```

Comment:
Is there a compelling reason to _not_ copy the pattern? _Technically_ one reason that copying the pattern is that the status object could contain additional details. But that's not likely to happen. Still, I see no good reason not to keep the status object as we always have.

Reply:
I'm actually thinking of taking it to the other direction and not even keeping the message from the wire. That way if the error is mutated (e.g. a different `Namespace` object is set, it is serialized correctly).

Reply:
I wouldn't put too much thought into it, just keep the status you're given like other similar errors

Reply:
Lol, I'll just do that to move things along. I wanted to improve what we had, but I can see arguments in both directions.

Reply:
Done.